docroot/modules/contrib/videojs/src/Plugin/Field/FieldFormatter/VideoJsPlayerListFormatter.php

```diff
@@ -51,27 +54,53 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
       $video_uri = $file->getFileUri();
       $video_items[] = Url::fromUri(file_create_url($video_uri));
     }
-    $elements[] = array(
-      '#theme' => 'videojs',
-      '#items' => $video_items,
-      '#player_attributes' => $this->getSettings(),
-      '#attached' => array(
-        'library' => array('videojs/videojs'),
-      ),
-    );
+    $var = $this->getSettings();
+    $extract_extensions = explode(',', $var['extensions']);
+    foreach ($extract_extensions as $key => $value) {
+      if (stripos($video_uri, $value)) {
+        
+        $ext = $value;
+      }
+      else {
+        $ext = 'mp4';
+      }
+    }
+    if ($this->getSetting('annotations') == 0) {
+      $elements[] = array(
+        '#theme' => 'videojs',
+        '#items' => $video_items,
+        '#player_extension' => $ext,
+        '#player_attributes' => $this->getSettings(),
+        '#attached' => array(
+          'library' => array('videojs/videojs'),
+        ),
+      );
+    }
+    else if($this->getSetting('annotations') == 1) {
+      $elements[] = array(
+        '#theme' => 'videojs',
+        '#items' => $video_items,
+        '#player_extension' => $ext,
+        '#player_attributes' => $this->getSettings(),
+        '#attached' => array(
+          'library' => array('videojs/videojs_annotation'),
+        ),
+      );
+    }
     return $elements;
   }
-  
+
   /**
    * {@inheritdoc}
    */
   public static function isApplicable(FieldDefinitionInterface $field_definition) {
     $entity_form_display = entity_get_form_display($field_definition->getTargetEntityTypeId(), $field_definition->getTargetBundle(), 'default');
     $widget = $entity_form_display->getRenderer($field_definition->getName());
     $widget_id = $widget->getBaseId();
-    if($field_definition->isList()){
+    if ($field_definition->isList()) {
       return TRUE;
     }
-    return FALSE;
+    return TRUE;
```

Comment:
Looks like this function is going to return TRUE in all the cases? Why do we need a condition here in that case?